src/localize.cpp

```diff
@@ -358,36 +331,61 @@ void localize_wrk(const BasisSet & basis, arma::mat & C, arma::vec & E, const ar
-      for(size_t io=0;io<orbidx.size();io++)
-	C.col(orbidx[io])=Cloc.col(io);
-      for(size_t io=0;io<orbidx.size();io++)
-	E(orbidx[io])=Eloc(io);
-    }
+      for(size_t io=0;io<C.n_cols;io++)
+	C.col(io)=Cloc.col(io);
+      for(size_t io=0;io<C.n_cols;io++)
+	E(io)=Eloc(io);
   }
 
-  // Compute size distribution
+   //Compute size distribution
   if(size) {
     if(start==UNITMAT)
-      size_distribution(basis,Cplx,sizedist,printidx);
+      size_distribution(basis,Cplx,sizedist,orbidx);
     else {
       arma::cx_mat Chlp=C*std::complex<double>(1.0,0.0);
-      size_distribution(basis,Chlp,sizedist,printidx);
-    }
+     size_distribution(basis,Chlp,sizedist,orbidx);
   }
 }
+}
+
+
+void localize(const BasisSet & basis, arma::mat & C, arma::vec & E, const arma::mat & P, const arma::mat & H, std::vector<double> occs, bool virt, enum locmet method, enum unitmethod umet, enum unitacc acc, enum startingpoint start, bool delocalize, std::string sizedist, bool size, std::string fname, double Gthr, double Fthr, int maxiter, unsigned long int seed, bool debug, int ncore, int nel) {
+
+  //TODO: simply function inputs (i.e. remove vars that are no longer used - occs etc.)
+
+    // Run localization, occupied core space
+  if(ncore) {
+    std::vector<size_t> core_idxs;
+    for (size_t i=0;i<ncore;i++)
+        core_idxs.push_back(i);
+
+    arma::subview<double> C_core = C.cols(0,ncore-1);
+	arma::subview_col<double> E_core = E.subvec(0,ncore-1);
+
+    localize_wrk(basis,C_core,E_core,P,H,core_idxs,method,umet,acc,start,delocalize,sizedist+".core",size,fname+".core",Gthr,Fthr,maxiter,seed,debug);
+  }
 
-void localize(const BasisSet & basis, arma::mat & C, arma::vec & E, const arma::mat & P, const arma::mat & H, std::vector<double> occs, bool virt, enum locmet method, enum unitmethod umet, enum unitacc acc, enum startingpoint start, bool delocalize, std::string sizedist, bool size, std::string fname, double Gthr, double Fthr, int maxiter, unsigned long int seed, bool debug) {
-  // Run localization, occupied space
-  localize_wrk(basis,C,E,P,H,occs,method,umet,acc,start,delocalize,sizedist+".o",size,fname+".o",Gthr,Fthr,maxiter,seed,debug);
+  std::vector<size_t> val_idxs;
+  for (size_t i=ncore;i<nel;i++)
+    val_idxs.push_back(i);
+
+  // Run localization, occupied valence space
+  arma::subview<double> C_occ = C.cols(0, nel-1);
+  arma::subview_col<double> E_occ = E.subvec(0,nel-1);
+
+  localize_wrk(basis,C_occ,E_occ,P,H,val_idxs,method,umet,acc,start,delocalize,sizedist+".val",size,fname+".val",Gthr,Fthr,maxiter,seed,debug);
 
   // Run localization, virtual space
   if(virt) {
-    for(size_t i=0;i<occs.size();i++)
-      if(occs[i]==0.0)
-	occs[i]=1.0;
-      else
-	occs[i]=0.0;
-    localize_wrk(basis,C,E,P,H,occs,method,umet,acc,start,delocalize,sizedist+".v",size,fname+".v",Gthr,Fthr,maxiter,seed,debug);
+    std::vector<size_t> virt_idxs;
+    for (size_t i=nel;i<C.n_cols;i++)
+      virt_idxs.push_back(i);
```

Comment:
Since the fractional occupations aren't used anymore, I don't see a need to carry the index vectors either.

Reply:
The only benefit of carrying the index vectors is for the size distribution file (i.e. to allow the global index to be printed in the size distribution file).

Reply:
The only benefit of carrying the index vector is to allow the global orbital index to be printed in the size distribution file. The `size_distribution` function is called within `localize_wrk`.

Reply:
Perhaps a solution is to call `size_distribution` from within `localize` (instead of `localize_wrk`), and modify 'size_distribution' to take stop and start index arguments (not a full index vector). This would make `localize_wrk` an index-agnostic function.

Reply:
Yeah, that would be a nicer workaround.